pkg/apis/kops/v1alpha2/componentconfig.go

```diff
@@ -760,6 +760,28 @@ type CloudConfiguration struct {
 	Openstack *OpenstackConfiguration `json:"openstack,omitempty"`
 }
 
+// ComponentConfig contains the configuration for kops components.
+type ComponentConfig struct {
+	// NodeTerminationHandlerConfig determines the cluster autoscaler configuration.
+	NodeTerminationHandler *NodeTerminationHandlerConfig `json:"nodeTerminationHandler,omitempty"`
+}
+
+// NodeTerminationHandlerConfig determines the cluster autoscaler configuration.
+type NodeTerminationHandlerConfig struct {
+	// Enabled enables the node termination handler.
+	// Default: false
```

Comment:
Should default to `true` when there is a non-`nil` `nodeTerminationHandler`

Reply:
Thanks. Will fix that.

Reply:
Do we have this behaviour anywhere else? Should we document and agree about it first at next office hours?

Reply:
CAS did not have this. I know kubeproxy does, but that one is a bit special. Either way is fine for me.

Reply:
kube-proxy has this behaviour because it is (historically) a required component.

Reply:
Precedence would say that it should be disabled by default even if the struct is non-nil. I'd also think that is the most intuitive default.

Reply:
I would disagree that it is the most intuitive default. Why would anyone specify a non-nil struct if they don't want that thing enabled? With the enabled field defaulting to false, specifying a non-nil struct and any other subfield will have no effect unless they also specify an explicit value for `enabled`.

Numerous existing things, such as `cloudControllerManager`, the authentication, authorization, networking, and API access options, etc. are enabled by specifying non-nil structs.

The point of an `enabled` field is to be able to disable a whole subcomponent/feature which has multiple config settings.

Reply:
OK, but let's document somewhere for the future that we expect this behaviour for new components.